components/export/api/src/main/java/org/phenotips/export/internal/DataToCellConverter.java

```diff
@@ -256,13 +256,39 @@ public DataSection phenotypeBody(Patient patient) throws Exception
         return section;
     }
 
+    public void variantsSetup(Set<String> enabledFields) throws Exception
+    {
+        String sectionName = "variants";
+        String[] fieldIds =
+        { "variants", "variants_genesymbol", "variants_interpretation", "variants_inheritance",
+        "variants_validated" };
+        // FIXME These will not work properly in different configurations
+        String[][] headerIds =
+        { { "hgvs_id" }, { "genesymbol", "hgvs_id" }, { "interpretation", "genesymbol", "hgvs_id" },
+            { "inheritance", "interpretation", "genesymbol", "hgvs_id" },
+            { "validated", "inheritance", "interpretation", "genesymbol", "hgvs_id" } };
+        Set<String> present = new HashSet<String>();
+
+        int counter = 0;
+        for (String fieldId : fieldIds) {
+            if (enabledFields.remove(fieldId)) {
+                for (String headerId : headerIds[counter]) {
+                    present.add(headerId);
+                }
+            }
+            counter++;
+        }
```

Comment:
This exact block is repeated 5 times in the file -- maybe move it into a private method?